extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3TaskLogs.java

```diff
@@ -87,38 +92,42 @@ public Optional<InputStream> streamTaskStatus(String taskid) throws IOException
   private Optional<InputStream> streamTaskFile(final long offset, String taskKey) throws IOException
```

Comment:
It might be nicer to just add a new method `streamTaskFileWithRetry`, which calls the existing `streamTaskFile`.

something like:
```java
private Optional<InputStream> streamTaskFileWithRetry(final long offset, String taskKey)
{
  try {
    return S3Utils.retryOperation(() -> streamTaskFile(offset, taskKey))
  }
  catch (Exception e) {
    throw new IOE(e, "Failed to stream logs from: %s", taskKey);
  }
}
```

Reply:
Note: the existing `streamTaskFile` method could throw some wrapped exception e.g. `new IOE(e, "Failed to stream logs from: %s", taskKey)`, in that case the outer new method `streamTaskFileWithRetry` won't know whether should retry or not. That means we need to refine the exception throwing in the existing `streamTaskFile`.

Reply:
Yeah, you can modify the exception if required, as long as the overall logic remains the same. The advantage of having a separate retry method is readability and a small diff.